corehq/blobs/tests/test_mixin.py

```diff
@@ -463,6 +466,239 @@ def setUpClass(cls):
         cls.db = PutInOldCopyToNewBlobDB(cls.db, TemporaryFilesystemBlobDB())
 
 
+class TestBlobHelper(BaseTestCase):
+
+    def setUp(self):
+        self.couch = FakeCouchDatabase()
+
+    def make_doc(self, type_=mod.BlobHelper, doc=None):
+        if doc is None:
+            doc = {}
+        doc["_id"] = uuid.uuid4().hex
+        if doc.get("_attachments"):
+            for name, attach in doc["_attachments"].iteritems():
+                self.couch.put_attachment(doc, name=name, **attach)
+        obj = type_(doc, self.couch)
+        if "external_blobs" in doc:
+            save_log = list(self.couch.save_log)
+            for name, attach in list(doc["external_blobs"].iteritems()):
+                obj.put_attachment(name=name, **attach)
+            self.couch.save_log = save_log
+        return obj
+
+    def test_put_and_fetch_attachment_from_couch(self):
+        obj = self.make_doc(doc={"_attachments": {}})
+        content = "test couch"
+        self.assertFalse(self.couch.data)
+        obj.put_attachment(content, "file.txt", content_type="text/plain")
+        self.assertEqual(obj.fetch_attachment("file.txt"), content)
+        self.assertTrue(self.couch.data)
+        self.assertFalse(self.couch.save_log)
+
+    def test_put_attachment_error_on_ambiguous_backend(self):
+        obj = self.make_doc()
+        with self.assertRaises(mod.AmbiguousBlobStorageError):
+            obj.put_attachment("test", "file.txt", content_type="text/plain")
+
+    def test_fetch_attachment_do_not_hit_couch_when_not_migrating(self):
+        def fetch_fail(*args, **kw):
+            raise Exception("fail!")
+        couch = FakeCouchDatabase()
+        couch.fetch_attachment = fetch_fail
+        obj = mod.BlobHelper({
+            "_id": "fetch-fail",
+            "external_blobs": {"not-found.txt": {"id": "hahaha"}},
+        }, couch)
+        self.assertFalse(obj.migrating_blobs_from_couch)
+        with self.assertRaises(mod.ResourceNotFound):
+            obj.fetch_attachment("not-found.txt")
```

Comment:
Not really following this test. I would have thought that it shoudn't raise an exception here